patches/server/0953-Fix-NPE-on-Allay-stopDancing-while-not-dancing.patch

```diff
@@ -0,0 +1,24 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: NonSwag <mrminecraft00@gmail.com>
+Date: Tue, 6 Dec 2022 23:04:21 +0100
+Subject: [PATCH] Fix NPE on Allay#stopDancing while not dancing
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftAllay.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftAllay.java
+index db3b0a9d3696f31d473342312d437e1eae97404a..29b25ee3081060b53358683a193196c01c80afb4 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftAllay.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftAllay.java
+@@ -82,8 +82,11 @@ public class CraftAllay extends CraftCreature implements org.bukkit.entity.Allay
+     @Override
+     public void stopDancing() {
+         this.getHandle().forceDancing = false;
+-        this.getHandle().jukeboxPos = null;
+-        this.getHandle().setJukeboxPlaying(null, false);
++        // Paper start - Fixing NPE
++        if (this.getHandle().jukeboxPos != null) {
++            this.getHandle().setJukeboxPlaying(this.getHandle().jukeboxPos, false);
++        }
++        // Paper end
```

Comment:
Why can't you just use setDancing(false) like the original fix?

Reply:
Forgot to add that back in, thanks.

Reply:
Overall switched to simpler logic

Reply:
Now the paper comments are too much :)

Reply:
There's no such thing as TOO many paper comments! 😄